includes/ajax-functions.php

```diff
@@ -192,35 +192,39 @@ function give_ajax_get_form_title() {
  * @return void
  */
 function give_ajax_get_states_field() {
+	$states_found = false;
+	// Get the Country code from the $_POST.
+	$country = sanitize_text_field( $_POST['country'] );
 
-	if ( empty( $_POST['country'] ) ) {
-		$_POST['country'] = give_get_country();
+	// Get the field name from the $_POST.
+	$field_name = sanitize_text_field( $_POST['field_name'] );
+
+	if ( empty( $country ) ) {
+		$country = give_get_country();
 	}
-	$states = give_get_states( $_POST['country'] );
 
+	$states = give_get_states( $country );
 	if ( ! empty( $states ) ) {
-
 		$args = array(
-			'name'             => $_POST['field_name'],
-			'id'               => $_POST['field_name'],
-			'class'            => $_POST['field_name'] . '  give-select',
-			'options'          => give_get_states( $_POST['country'] ),
+			'name'             => $field_name,
+			'id'               => $field_name,
+			'class'            => $field_name . '  give-select',
+			'options'          => $states,
 			'show_option_all'  => false,
 			'show_option_none' => false,
 		);
-
-		$response = Give()->html->select( $args );
-
+		$data = Give()->html->select( $args );
+		$states_found = true;
 	} else {
-
-		$response = 'nostates';
+		$data = 'nostates';
 	}
-
-	echo $response;
-
-	give_die();
+	$response = array(
+		'success' => true,
+		'states_found' => $states_found,
+		'data' => $data,
+	);
+	wp_send_json( $response );
```

Comment:
@raftaar1191 I agree as we discuss on call. As per current solution, state field will not print as text field as it was happening before. We can add a filter here to show/hide text field on basis of state if admin wants by default we can hide text field for `Germany`, `UAE` etc.